### DISCUSSION TITLE
tileContent does not load under the vite tool

### DISCUSSION THREAD

**Discussion by peiyanlu** [iTwin/itwinjs-core#5185] [Q&A] [answered]
When I use the vite tool, it behaves differently than webpack, tileContent-related requests are not issued, what should I do to make them behave consistently.The following are the specific requests under different build tools, and some requests are missing when using VITE

<img width="599" alt="Snipaste_2023-03-10_13-54-17" src="https://user-images.githubusercontent.com/37688666/224234871-5a2e6b45-96f9-40b2-b956-11fbdc340414.png">
<img width="634" alt="Snipaste_2023-03-10_13-55-52" src="https://user-images.githubusercontent.com/37688666/224235113-58627687-e4f0-44af-bf4a-7bc344b12d1c.png">

**Comment by aruniverse** [comment, marked as answer]
Great to hear you're using vite!

We've noticed an issue, rather a difference with how vite works in development vs production builds, and how it resolves dynamic imports. One uses esbuild and the other uses rollup if I'm not mistaken, but why there's a difference not sure. 

Can you share the version of vite you're using? And maybe also your vite config?

If you use something like [pnpm patch](https://pnpm.io/cli/patch) you can apply the following patch and tiles should start loading on screen.

```diff
diff --git a/lib/esm/tile/TileAdmin.js b/lib/esm/tile/TileAdmin.js
index 4e1a0bcfa3b8e90137e887503476c26d1a293ecd..47afddda8a4859e15457966633facc0fbe05efdf 100644
--- a/lib/esm/tile/TileAdmin.js
+++ b/lib/esm/tile/TileAdmin.js
@@ -155,7 +155,8 @@ export class TileAdmin {
         this._tileStoragePromise = (async () => {
             await import("reflect-metadata");
             // eslint-disable-next-line @typescript-eslint/naming-convention
-            const { AzureFrontendStorage, FrontendBlockBlobClientWrapperFactory } = await import(/* webpackChunkName: "object-storage" */ "@itwin/object-storage-azure/lib/frontend");
+            const objectStorageImport = await import("@itwin/object-storage-azure/lib/frontend");
+            const { AzureFrontendStorage, FrontendBlockBlobClientWrapperFactory } = objectStorageImport.default ? objectStorageImport.default : objectStorageImport;
             const azureStorage = new AzureFrontendStorage(new FrontendBlockBlobClientWrapperFactory());
             this._tileStorage = new TileStorage(azureStorage);
             return this._tileStorage;
```


Another option is to add the following plugin to your vite config:


```ts
class StringReplacePlugin implements Plugin {
  public name = StringReplacePlugin.name;
  public enforce = "pre" as const;

  public transform(code: string, _id: string) {
    return code.replace(
      /const { AzureFrontendStorage, FrontendBlockBlobClientWrapperFactory } = await import\((.+?)\);/s,
      `
      const objectStorage = await import($1);
      const { AzureFrontendStorage, FrontendBlockBlobClientWrapperFactory } = objectStorage.default ?? objectStorage;
      `,
    );
  }
```

**Reply by peiyanlu** [reply]
Thank you very much, the method provided is very useful. Before that, I had no other configuration other than a request proxy
